async_processor/amqp_pub_sub.py

```diff
@@ -0,0 +1,107 @@
+from contextlib import asynccontextmanager
+from typing import AsyncIterator, Optional
+
+from aio_pika import Message, connect_robust
+from aio_pika.exceptions import AMQPError, QueueEmpty
+
+from async_processor.types import (
+    AMQPInputConfig,
+    AMQPOutputConfig,
+    Input,
+    InputFetchAckFailure,
+    InputMessageFetchFailure,
+    Output,
+)
+
+
+class AMQPInput(Input):
+    def __init__(self, config: AMQPInputConfig):
+        self._queue_url = config.queue_url
+        self._queue_name = config.queue_name
+        self._connection = None
+
+    async def _connect(self):
+        self._connection = await connect_robust(self._queue_url)
+
+    async def _disconnect(self):
+        if self._connection is not None:
+            await self._connection.close()
+
+    @asynccontextmanager
+    async def get_input_message(
+        self,
+    ) -> AsyncIterator[Optional[str]]:
+        try:
+            await self._connect()
```

Comment:
I feel that the connection should be in constructor itself and then used accross

Reply:
I have tried that one, its not working
AttributeError: 'coroutine' object has no attribute 'channel'
do you have another workaround?